jwst/outlier_detection/outlier_detection_ifu.py

```diff
@@ -53,203 +49,258 @@ def __init__(self, input_models, reffiles=None, **pars):
-        resampled_wht = [i.weightmap for i in resampled_models]
-        nlow = self.outlierpars.get('nlow', 0)
-        nhigh = self.outlierpars.get('nhigh', 0)
-        maskpt = self.outlierpars.get('maskpt', 0.7)
-        badmasks = []
-        for w in resampled_wht:
-            # Due to a bug in numpy.nanmean, need to check
-            # for a completely zero array
-            if not np.any(w):
-                mean_weight = 0.
-            else:
-                mean_weight, _, _ = sigma_clipped_stats(
-                    w, sigma=3.0, mask_value=0.
-                )
-            weight_threshold = mean_weight * maskpt
-            # Mask pixels were weight falls below MASKPT percent of
-            #    the mean weight
-            mask = np.less(w, weight_threshold)
-            log.debug("Number of pixels with low weight: {}".format(
-                np.sum(mask)))
-            badmasks.append(mask)
-
-        # Compute median of stack os images using BADMASKS to remove low weight
-        # values
-        median_image = median(resampled_sci, nlow=nlow, nhigh=nhigh,
-                              badmasks=badmasks)
-        # for i in range(len(resampled_sci)):
-        #    data = resampled_sci[i]
-        #    mask = badmasks[i]
-        return median_image
-
-    def blot_median(self, median_image):
-        """IFU-specific version of blot_median."""
-        cubeblot = blot_cube_build.CubeBlot(median_image, self.input_models)
-        cubeblot.blot_info()
-        blot_models, input_list_number = cubeblot.blot_images()
-        for j in range(len(blot_models)):
-            k = input_list_number[j]
-            self.blot_models[k].data += blot_models[j].data
-            self.blot_models[k].meta = blot_models[j].meta
-
-
-class ErrorWrongInstrument(Exception):
-    """ Raises an exception if the instrument is not MIRI or NIRSPEC
-    """
-    pass
+            detector_name = uq_det[idet]
+            opt_info = (kern_size[0], kern_size[1], threshold_percent,
+                        diffarr, minarr, normarr, minarr_norm)
+            opt_model = self.create_optional_results_model(opt_info)
+            opt_model.meta.filename = self.make_output_path(
+                basepath=self.input_models.meta.asn_table.products[0].name,
+                suffix=detector_name + '_outlier_output')
+            log.info("Writing out intermediate outlier file {}".format(opt_model.meta.filename))
+            opt_model.save(opt_model.meta.filename)
+
+        del diffarr
+
+        # store some information if the second flagging step is to be done. 
+        if ifu_second_check:
+            # store where the minarr is nan (neighbor pixels have nan so differences produces a nan)
+            nanminarr = np.isnan(minarr)
+            nanindx = np.where(nanminarr)
+
+        # Update DQ flag
+        for i in range(len(self.input_models)):
+            model = self.input_models[i]
+            sci = model.data
+            dq = model.dq
```

Comment:
It looks like these are worked with only if the detector matches, should be in the `if` statement.

Reply:
done